### PR TITLE
Removing temporary tables, trimming names

### PR DESCRIPTION
**Pull request by bipin-rudder**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: warehouse/postgres/postgres.go [hunk 1]
```diff
@@ -344,8 +357,11 @@ func (pg *HandleT) loadUserTables() (errorMap map[string]error) {
 	}
 	errorMap[warehouseutils.UsersTable] = nil
 
-	unionStagingTableName := misc.TruncateStr(fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, strings.Replace(uuid.NewV4().String(), "-", "", -1), "users_identifies_union"), 127)
-	stagingTableName := misc.TruncateStr(fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, strings.Replace(uuid.NewV4().String(), "-", "", -1), warehouseutils.UsersTable), 127)
+	unionStagingTableName := misc.TruncateStr(fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, strings.Replace(uuid.NewV4().String(), "-", "", -1), "users_identifies_union"), 63)
+	stagingTableName := misc.TruncateStr(fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, strings.Replace(uuid.NewV4().String(), "-", "", -1), warehouseutils.UsersTable), 63)
+	defer pg.dropStagingTable(stagingTableName)
+	defer pg.dropStagingTable(unionStagingTableName)
+	defer pg.dropStagingTable(identifyStagingTable)
```
Comment (psrikanth88) [inline warehouse/postgres/postgres.go:364]: we are returning on error above in one of the cases. Should we have this drop there too?

File: warehouse/postgres/postgres.go [hunk 2]
```diff
@@ -191,6 +191,12 @@ func (pg *HandleT) DownloadLoadFiles(tableName string) ([]string, error) {
 }
 
 func (pg *HandleT) loadTable(tableName string, tableSchemaInUpload warehouseutils.TableSchemaT, skipTempTableDelete bool) (stagingTableName string, err error) {
+	sqlStatement := fmt.Sprintf(`SET search_path to "%s"`, pg.Namespace)
```
Comment (psrikanth88) [inline warehouse/postgres/postgres.go:194]: should we use search_path or just mention schema name in create table query ? you mentioned somthing about it not using search path in few cases even though it was set, right?

File: warehouse/postgres/postgres.go [hunk 3]
```diff
@@ -209,8 +215,8 @@ func (pg *HandleT) loadTable(tableName string, tableSchemaInUpload warehouseutil
 		return
 	}
 	// create temporary table
-	stagingTableName = fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, tableName, strings.Replace(uuid.NewV4().String(), "-", "", -1))
-	sqlStatement := fmt.Sprintf(`CREATE TEMPORARY TABLE "%[2]s" (LIKE "%[1]s"."%[3]s")`, pg.Namespace, stagingTableName, tableName)
+	stagingTableName = misc.TruncateStr(fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, tableName, strings.Replace(uuid.NewV4().String(), "-", "", -1)), 63)
```
Comment (psrikanth88) [inline warehouse/postgres/postgres.go:218]: can you also make this change in rudder-transformer to handle cases where the table and column names are > 63

**Review by psrikanth88 — CHANGES_REQUESTED** [review]

File: warehouse/postgres/postgres.go (hunk 1 above)
Comment (bipin-rudder) [reply warehouse/postgres/postgres.go]: makes sense. changed accordingly.

File: warehouse/postgres/postgres.go (hunk 3 above)
Comment (bipin-rudder) [reply warehouse/postgres/postgres.go:218]: made the transformer changes accordingly:
https://github.com/rudderlabs/rudder-transformer/pull/572/files

**Pull request merged**